include/swift/SIL/SILNode.h

```diff
@@ -92,62 +92,93 @@ static_assert(unsigned(SILNodeKind::Last_SILNode) < (1 << NumSILNodeKindBits),
 ///   ValueBase subobject, the cast will yield a corrupted value.
 ///   Always use the LLVM casts (cast<>, dyn_cast<>, etc.) instead.
 class alignas(8) SILNode {
+public:
+  /// The assumed number of bits that a SILNode+padding will take up. Public for
+  /// static assertion purposes.
+  static constexpr unsigned NumTotalSILNodeBits = 64;
+
 protected:
-  enum class SILNodeStorageLocation {
-    Value,
-    Instruction
+  static constexpr unsigned NumKindBits = NumSILNodeKindBits;
+  static constexpr unsigned NumStorageLocBits = 1;
+  static constexpr unsigned NumIsRepresentativeBits = 1;
+  static constexpr unsigned NumSubclassDataBits =
+      NumTotalSILNodeBits - NumKindBits - NumStorageLocBits -
+      NumIsRepresentativeBits;
+
+  enum class SILNodeStorageLocation : uint8_t { Value, Instruction };
+
+  enum class IsRepresentative : uint8_t {
+    No = 0,
+    Yes = 1,
   };
 
 private:
-  const unsigned Kind : NumSILNodeKindBits;
-  const unsigned StorageLoc : 1;
-  const unsigned IsCanonical : 1;
-
-  // TODO: Pack other things in here.
+  const unsigned Kind : NumKindBits;
+  const unsigned StorageLoc : NumStorageLocBits;
+  const unsigned IsRepresentativeNode : NumIsRepresentativeBits;
+  uint64_t SubclassData : NumSubclassDataBits;
```

Comment:
In order to pack these effectively on all platforms, please make them all uint64_t.